### PR TITLE
HBASE-25003 Backport HBASE-24350 and HBASE-24779 to branch-2.2

### PR DESCRIPTION
**Pull request by tamasadami**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  7s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 12s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 34s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 48s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  5s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 10s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 21s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 17s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 31s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 31s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 15s |  hbase-server: The patch generated 4 new + 10 unchanged - 0 fixed = 14 total (was 10)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  2s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m  9s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   1m  9s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 38s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 35s |  hbase-hadoop-compat in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 40s |  hbase-hadoop2-compat in the patch passed.  |
| +1 :green_heart: |  unit  | 117m  9s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 27s |  The patch does not generate ASF License warnings.  |
|  |   | 186m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.13 Server=19.03.13 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2565 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 5dd580e4b0a2 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2565/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 06d5aadd92 |
| Default Java | 1.8.0_181 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/1/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/1/testReport/ |
| Max. process+thread count | 3359 (vs. ulimit of 12500) |
| modules | C: hbase-hadoop-compat hbase-hadoop2-compat hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 50s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 35s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 30s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 51s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 27s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  6s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 38s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 32s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 13s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 31s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 31s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  The patch passed checkstyle in hbase-hadoop-compat  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  The patch passed checkstyle in hbase-hadoop2-compat  |
| +1 :green_heart: |  checkstyle  |   1m 29s |  hbase-server: The patch generated 0 new + 9 unchanged - 1 fixed = 9 total (was 10)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 23s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  27m 45s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   1m  5s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 53s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 33s |  hbase-hadoop-compat in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 38s |  hbase-hadoop2-compat in the patch passed.  |
| +1 :green_heart: |  unit  | 269m 25s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 16s |  The patch does not generate ASF License warnings.  |
|  |   | 345m  9s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.13 Server=19.03.13 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2565 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 0042a63064e9 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2565/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 24265c56fe |
| Default Java | 1.8.0_181 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/2/testReport/ |
| Max. process+thread count | 2674 (vs. ulimit of 12500) |
| modules | C: hbase-hadoop-compat hbase-hadoop2-compat hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsSource.java [hunk 1]
```diff
@@ -119,7 +146,7 @@ public long getLastTimeStampOfWalGroup(String walGroup) {
    * @param walGroup which group we are getting
    * @return age
    */
-  public long getAgeofLastShippedOp(String walGroup) {
+  public long getAgeOfLastShippedOp(String walGroup) {
```
Comment (wchevreuil) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsSource.java:149]: nit: Unfortunately, we can't fix this method naming typo on IA limited private for a maintenance release.

**Review by wchevreuil — COMMENTED** [review]
LGTM overall, just need to revert back a public method rename in MetricsSource, since it's LimitedPrivate.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsSource.java (hunk 1 above)
Comment (tamasadami) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsSource.java]: Good catch. I have just pushed to fix for that.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 19s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m 33s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 45s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  9s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 12s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 19s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 14s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 56s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 32s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 32s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  The patch passed checkstyle in hbase-hadoop-compat  |
| +1 :green_heart: |  checkstyle  |   0m 15s |  The patch passed checkstyle in hbase-hadoop2-compat  |
| +1 :green_heart: |  checkstyle  |   1m 17s |  hbase-server: The patch generated 0 new + 9 unchanged - 1 fixed = 9 total (was 10)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  4s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m 14s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   1m  9s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   4m 37s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 35s |  hbase-hadoop-compat in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 43s |  hbase-hadoop2-compat in the patch passed.  |
| +1 :green_heart: |  unit  | 150m  4s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 31s |  The patch does not generate ASF License warnings.  |
|  |   | 218m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.13 Server=19.03.13 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2565 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux eb485c8fb159 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2565/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 2e244a3bdc |
| Default Java | 1.8.0_181 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/3/testReport/ |
| Max. process+thread count | 3457 (vs. ulimit of 12500) |
| modules | C: hbase-hadoop-compat hbase-hadoop2-compat hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2565/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by infraio** [comment]
Seems these patch not go into branch-2.3, too? I thought these should go into 2.3 firstly.

**Comment by tamasadami** [comment]
> Seems these patch not go into branch-2.3, too? I thought these should go into 2.3 firstly.

Yes, we have an open PR against branch-2.3 as well: https://github.com/apache/hbase/pull/2374

**Review by wchevreuil — APPROVED** [review]

**Pull request merged**